tests/Unit/Core/Domain/Customer/ValueObject/CustomerIdTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace Tests\Unit\Core\Domain\Customer\ValueObject;
+
+use Generator;
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Exception\CustomerConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Customer\ValueObject\CustomerId;
+
+class CustomerIdTest extends TestCase
+{
+    /**
+     * @dataProvider getValidValues
+     *
+     * @param int $value
+     */
+    public function testCreateWithPositiveValue(int $value): void
+    {
+        $customerId = new CustomerId($value);
+
+        $this->assertEquals($value, $customerId->getValue());
+    }
+
+    /**
+     * @dataProvider getInvalidValues
+     *
+     * @param int $value
+     */
+    public function testItThrowsExceptionWhenProvidingInvalidValue(int $value): void
+    {
+        $this->expectException(CustomerConstraintException::class);
+        $this->expectExceptionCode(CustomerConstraintException::INVALID_ID);
+
+        new CustomerId($value);
+    }
+
+    /**
+     * @return Generator
+     */
+    public function getValidValues(): Generator
+    {
+        yield [
+            0,
```

Comment:
`CustomerId` should not accept 0 as a valid value

Reply:
well it does. 
Old implementation was done to accept that and Im not messing with it in this PR as it affects much bigger scope if I try changing that